src/Core/Grid/Presenter/AccessbilityChecker/Row/EmployeeDeleteAccessibilityChecker.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Presenter\AccessbilityChecker\Row;
+
+use PrestaShop\PrestaShop\Core\Grid\Action\Row\RowActionInterface;
+
+/**
+ * Checks if "Delete" action can be performed on employee
+ */
+final class EmployeeDeleteAccessibilityChecker implements RowAccessibilityCheckerInterface
+{
+    /**
+     * @var int
+     */
+    private $contextEmployeeId;
+
+    /**
+     * @param int $contextEmployeeId
+     */
+    public function __construct($contextEmployeeId)
```

Comment:
Yes one thousand times 😄 that would help us get rid of things like https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Resources/config/services/adapter/meta.yml#L18 that I feel very unsecure about because it can break so easily without easy error message.

But this is a huge work here, can you create an issue for it ? And I guess it will be for 1.7.7 since 1.7.6 is already overloaded.